### PR TITLE
uvcdat script now will redirect all output appropriately

### LINKED ISSUE #1239: Redirect all (i.e. non-Python) application output to log file
Right now, all output from Python code, on `sys.stdout`/`sys.stderr`, is redirected to the log file.

Some people seem to be now arguing in favor of redirecting at the OS-level, so that no output would ever reach the terminal.

Pros:
* No output in terminal
* If some of that output is important, it would be in the log that would be submitted on error (I don't think any of it is important)

Cons:
* Ugly non-portable system hack
* Not Python
* Unexpected madness; the system prints diagnostics messages related to UI toolkits, we shouldn't be messing with that
* Clogging the logfile: these messages are often warnings from the UI toolkit or Qt, about mismatching fonts and compatiblity modes, which are completely useless to diagnose application-level errors
* This is not what anybody does. Run any random Linux application from a terminal, you will see tons of warnings; why is this more of a problem to us than anybody else?
* Warning/debug output from UV-CDAT components should be logged the right way. Leftover debug code should be dealt with, and will be annoying when running scripts with no recourse
* Again, ugly non-Python system hack

For the record, I am also highly opposed to redirecting `sys.stdout`/`sys.stderr` in the first place (just use Python's convenient logging facilities).

### PR DESCRIPTION
**Pull request by chaosphere2112**
Fixes #1239; parses arguments passed to UV-CDAT, detects where output should be sent, and redirects all stdout and stderr to append at that location (in addition to the standard UV-CDAT logging mechanisms that are already printing at that point). Should fix the issues we've been having with miscellaneous C++ level logging getting dumped to the console (like the annoying Qt modal window non-error).

### CODE REVIEW

**Comment by aashish24** [comment]
@chaosphere2112 as part of this PR could you please add documentation as well? Wiki for now should be good.

**Comment by chaosphere2112** [comment]
@aashish24 Documentation for what? The command line args we accept?

**Comment by aashish24** [comment]
> @aashish24 Documentation for what? The command line args we accept?

Right.

**Comment by aashish24** [comment]
If it makes sense.

**Comment by chaosphere2112** [comment]
@aashish24 Well, they do get displayed if you do `--help`, but I can stick that onto the wiki.

**Comment by aashish24** [comment]
@chaosphere2112 thanks.

**Comment by aashish24** [comment]
@chaosphere2112 I meant for capture, target etc. as well but sounds like they are shown as well?

File: CMake/cdat_modules_extra/uvcdat.in [hunk 1]
```diff
@@ -1,4 +1,40 @@
...
+
+    case $var in
+    -o) capture=true;
+            ;;
+    --output=*) target=`sed "s/--output=\(.*\)/\1/" <<< $var`
+        if [ "$target" = "" ]; then
+            redirect=false
+        fi
+            ;;
+        --output-std) redirect=false
```
Comment (aashish24) [inline CMake/cdat_modules_extra/uvcdat.in]: is this mis-aligned?

**Comment by aashish24** [comment]
@chaosphere2112 looks good to me. I didn't run it myself. Just had one minor question.

**Comment by chaosphere2112** [comment]
Ah, I misunderstood. Added some comments to the script to explain what everything does.

**Comment by aashish24** [comment]
thanks for taking care of it @chaosphere2112. LGTM :+1:

**Pull request merged**